plugins/discovery-ec2/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -42,4 +42,13 @@ grant {
   permission java.net.SocketPermission "*", "connect";
 
   permission java.lang.reflect.ReflectPermission "suppressAccessChecks";
+
+  permission java.util.PropertyPermission "aws.sharedCredentialsFile", "read,write";
```

Comment:
I think you only need `read,write` for `sharedCredentialsFile` and `configFile`, others are read only, right?

Reply:
We set those in tests. There's actually an `  // only for tests : org.opensearch.repositories.s3.S3RepositoryPlugin` in there. Should we move this stuff into build.gradle permissions hacking?

Reply:
Right, we supply this `plugin-security.policy` for production, we shouldn't add something for a test here

Reply:
https://github.com/opensearch-project/OpenSearch/pull/7372/commits/ace17645b894c5b7bf6874f127580b8240eaf534